lib/rubygems/bundler_version_finder.rb

```diff
@@ -3,15 +3,6 @@
 require "rubygems/util"
 
 module Gem::BundlerVersionFinder
-  @without_filtering = false
-
-  def self.without_filtering
-    without_filtering, @without_filtering = true, @without_filtering
```

Comment:
So, I have no idea what this "filtering" intends to do, but I think the problem is that here the assignment is inverted (it should be `without_filtering, @without_filtering = @without_filtering, true`. That means that once this "filtering" goes back to true, it's never switched back. Since this filtering is used further below in the `bundler_version_with_reason` to prevent the method from returning an useful message, that's what it ends up happening.

So, I think this could also be fixed by inverting the assignment but I wonder what we're trying to achieve with all this untested code... :man_shrugging:.